cocos/2d/CCNode.cpp

```diff
@@ -1236,29 +1191,32 @@ const kmMat4& Node::getNodeToParentTransform() const
 
         // Build Transform Matrix
         // Adjusted transform calculation for rotational skew
-        kmScalar mat[] = { cy * _scaleX, sy * _scaleX,     0,  0,
-                        -sx * _scaleY, cx * _scaleY,    0,  0,
+        _transform = { cy * _scaleX, sy * _scaleX, 0,  0,
+                      -sx * _scaleY, cx * _scaleY, 0,  0,
                         0,  0,  1,  0,
                         x,  y,  0,  1 };
 
-        kmMat4Fill(&_transform, mat);
         // XXX: Try to inline skew
         // If skew is needed, apply skew and then anchor point
         if (needsSkewMatrix)
         {
-            kmMat4 skewMatrix = {    1, tanf(CC_DEGREES_TO_RADIANS(_skewY)), 0, 0,
-                                    tanf(CC_DEGREES_TO_RADIANS(_skewX)),1, 0, 0,
-                                    0,  0, 1, 0,
-                                    0,  0,  0, 1};
+            kmMat4 skewMatrix = { 1, tanf(CC_DEGREES_TO_RADIANS(_skewY)), 0, 0,
+                                  tanf(CC_DEGREES_TO_RADIANS(_skewX)), 1, 0, 0,
+                                  0,  0,  1, 0,
+                                  0,  0,  0, 1};
 
             kmMat4Multiply(&_transform, &skewMatrix, &_transform);
 
             // adjust anchor point
             if (!_anchorPointInPoints.equals(Point::ZERO))
             {
                 // XXX: Argh, kmMat needs a "translate" method
-                _transform.mat[12] += -_anchorPointInPoints.x;
-                _transform.mat[13] += -_anchorPointInPoints.y;
+                kmVec4 v = {-_anchorPointInPoints.x, -_anchorPointInPoints.y, 0, 1};
+                kmVec4Transform(&v, &v, &_transform);
+                _transform.mat[12] = v.x;
+                _transform.mat[13] = v.y;
+                _transform.mat[14] = v.z;
+                _transform.mat[15] = v.w;
```

Comment:
@ricardoquesada ,Sure, I found that the behavior of skew in 2.x & 3.x is different.
In 2.x, setSkewX(30) + setRotation(30) will show:
<img width="959" alt="2 x" src="https://cloud.githubusercontent.com/assets/493372/19631662/b7a029bc-99d1-11e6-9f4f-b879cf1b315e.png">

In 3.x, setSkewX(30) + setRotation(30) will show:
<img width="957" alt="3 x" src="https://cloud.githubusercontent.com/assets/493372/19631664/ba91e7a0-99d1-11e6-853e-e3e8ed94eaeb.png">

TEST CODE:

2.x:

```c++
    {
        auto sprite = CCSprite::create("fonts/grossini.png");
        sprite->setPosition(ccp(visibleSize.width/2 + origin.x, visibleSize.height/2 + origin.y));
        addChild(sprite, 8888);
    }

    {
        auto sprite = CCSprite::create("fonts/grossini.png");
        sprite->setPosition(ccp(visibleSize.width/2 + origin.x, visibleSize.height/2 + origin.y));

        sprite->setSkewX(30);// SET SKEWX
        sprite->setRotation(30);// SET ROTATION

        addChild(sprite, 99999);
    }

    {
        CCDrawNode* d = CCDrawNode::create();

        d->drawSegment(ccp(origin.x, origin.y + visibleSize.height/2), ccp(origin.x + visibleSize.width, origin.y + visibleSize.height/2), 0.5f, ccc4f(1.0f, 0, 0, 1.0f));
        d->drawSegment(ccp(origin.x + visibleSize.width/2, origin.y), ccp(origin.x + visibleSize.width/2, origin.y + visibleSize.height), 0.5f, ccc4f(1.0f, 0, 0, 1.0f));

        addChild(d);
    }
```
3.x:
```c++
    {
        auto sprite = Sprite::create("fonts/grossini.png");
        sprite->setPosition(Vec2(visibleSize / 2) + origin);

        addChild(sprite, 8888);
    }

    {
        auto sprite = Sprite::create("fonts/grossini.png");
        sprite->setPosition(Vec2(visibleSize / 2) + origin);

        sprite->setSkewX(30); // SET SKEWX 
        sprite->setRotation(30); // SET ROTATION

        addChild(sprite, 99999);
    }

    {
        auto d = DrawNode::create();
        d->drawLine(Vec2(origin.x, origin.y + visibleSize.height/2), Vec2(origin.x + visibleSize.width, origin.y + visibleSize.height/2), Color4F::RED);
        d->drawLine(Vec2(origin.x + visibleSize.width/2, origin.y), Vec2(origin.x + visibleSize.width/2, origin.y + visibleSize.height), Color4F::RED);

        addChild(d);
    }
```

I don't know which one should be the correct one. At glance, I figure that the behavior of `setSkewX` + `setRotation` is correct in 2.x but wrong in 3.x.

In addition, cocos2d-html5 seems to use the behavior in 2.x, so some `skew + rotation` animations will be broken in 3.x while they work in cocos2d-x 2.x & cocos2d-html5.